### ISSUE TITLE
Future Plans? Maintainer contact?

### ISSUE THREAD

**Issue by achimmihca** [UltraStar-Deluxe/Play#37]
I see UltraStar Play based on Unity as a very neat idea. I tried it with current Unity 2019.02. It is working. Sadly there is not much yet.

Are there active developers or developers that want to contribute? The Gitter seems abandoned. Is there any chat or similar (e.g. Discord)?

Where do you developers live? 
I am from Germany and I think you as well. So you can answer in German if you please.

**Comment by rooi** [comment]
I would like to contribute, but the lack of activity is not very motivating.

**Comment by basisbit** [comment]
tldr: PLEASE join this project so we can finally do awesome karaoke stuff without having to deal with the old UltraStar Deluxe code base :smile: 

Hi there! This project is very much "not dead"... I just wasn't able to keep focused/motivated to learn Unity UI related stuff. Personally I am a backend developer (Java+Spring+TypeScript) and I guess this is why I tend to not like frontend stuff so much :sweat_smile:. Anyways, please read the [CONTRIBUTING.md](https://github.com/UltraStar-Deluxe/Play/blob/master/CONTRIBUTING.md) for a start. As soon as some more people join this project, I'll definitely be motivated to regularly spend time on this nice project and work together with all of you.
I tried to establish the Gitter chat within the USDX project and also this project here, but barely anyone used this chat platform. Gitter allows for read-only access to the chat if you are not logged it, and write-access if you log in for example using your Github account. If you all favor Discord, we can create a group there.

@achimmihca I also live in Germany (Ruhrgebiet), but at least for github issues (as documentation) we should stick to english. Of course we can chat in German, but somehow have to make sure we don't exclude the majority of the world from our discussions ;-)

**Comment by nikolaosginos** [comment]
I'd be glad to contribute as good as I can as well if this whole thing get's going.

Gitter seems suitable but I could also use Discord.

I am from Germany as well but I also think that we should stick to English if possible. 😅

Do you think we should try to structure and prioritize the issues? The project is in a very early state and we need to get the basics working I guess.

I am asking because I am not sure myself which parts of the game one should focus on right now. 😅

**Comment by achimmihca** [comment]
Nice to see others are interested in the project ;)

I think what is needed is a minimal usable application state. This includes:

- Reading UltraStar txt files
- Starting the song
- -> play the music
- -> play the video in sync with music (mind the GAP etc.)
- -> display notes in sync with the music
- -> display lyrics in sync with the music

After this works, one can add players
- having one player with a mic
- map the recorded sound to a note
- record the played note and display it in sync with the music

After this works, one can add scores and options and everything else.

Actually, my motivation is to develop is a more advanced UltraStar song file editor. A good data model for UltraStar songs and recording and playback is the basis for this too. This is why I thought that I could re-use UltraStar Play as my basis.

I will try to implement the core features from above. Adding multiple players, duetts, profiles, scores, game modes, persistence and so on is future work and not my focus.

**Comment by basisbit** [comment]
reading and parsing UltraStar text files is already done. Just playing audio is easy to do. The difficults/work will rather be to think of how to keep audio playback, notes displaying, audio-input-data and video playback in sync.
I guess we'll need some basic UI where the user can choose a song and then start the playback. I haven't figured out how to properly create any UI elements from code from fabrics (without using super slow reflection), how to position them so that it works for any aspect ratio and so on.

**Comment by basisbit** [comment]
(sorry I wasn't able to do any work this weekend. How to do emergency support for a customer all weekend)

**Comment by basisbit** [comment]
reopened this issue so we don't forget to adapt/create documentation

**Comment by Baklap4** [comment]
Hey all
Regarding communication:
* is there any communication set up yet and documented?
** There was/is a gitter chat but do we still want to use it? Other options being irc, discord, or none.
* Is there a public website?
** This could contain a dev blog
** Info about the project
** How to set stuff up etc
** Can be managed within a git repo and hosted through github.io
* Is there a mail address which can be used?
** For additional support and contact

This all would create a more robust community around UltraStar - Play

**Comment by basisbit** [comment]
- communication: there is gitter, but people seem to not like / adopt it. I was considering setting up a discord server, or whatever the currently most used group chat solution is. @achimmihca what is your opinion on this?
- usplay.net was planned to be used as website, but so far no one started setting up anything. Indeed, a jekyll based news blog would be nice. But then again, the usdx.eu website is not really used for anything, either. Eventually, it would be nice to also host a leaderboard on usplay.net . If anyone wants to set this up, please send me a message and I'll set up whatever repository or similar that is needed.
- In the ultrastar deluxe project, there exists a public email address, and that only collects spam, people submitting feature requests or people asking for where to get songs. I am not sure if we want a contact email address on that website except for legal reasons. Please feel free to change my opinion. 😛

**Comment by Baklap4** [comment]
> communication: there is gitter, but people seem to not like / adopt it. I was considering setting up a discord server, or whatever the currently most used group chat solution is. @achimmihca what is your opinion on this?

Personally discord is actively used just as slack is. Discord is more for gaming while on the other hand slack more 'professional' but lacks some features found in Discord.

> usplay.net was planned to be used as website, but so far no one started setting up anything. Indeed, a jekyll based news blog would be nice. But then again, the usdx.eu website is not really used for anything, either. Eventually, it would be nice to also host a leaderboard on usplay.net . If anyone wants to set this up, please send me a message and I'll set up whatever repository or similar that is needed.

Hmmm I might be available to create such a website. For now i'll just work local and if it's production worthy it can be shared :) Any features you'd like to see?
* Blogs (static pages or db?)
* Faq
* Some info regarding building/supporting this project
* please extend the list

> In the ultrastar deluxe project, there exists a public email address, and that only collects spam, people submitting feature requests or people asking for where to get songs. I am not sure if we want a contact email address on that website except for legal reasons. Please feel free to change my opinion.

Can't change your mind right now ;)

**Comment by achimmihca** [comment]
> @achimmihca what is your opinion on this?

I don't have experience with Discord, Gitter, nor Slack, sorry. I guess any chat solution will do if it is actually beeing used.

**Comment by achimmihca** [comment]
Ohh wow, I opened this issue already half a year ago! My original question has been answered and there has been quite some progress with the project.

Shall we close this issue?

**Comment by basisbit** [comment]
I agree. By the way, there now exists a first basic draft of the website. Content can be easily written in markdown and be pull requested to https://github.com/UltraStar-Deluxe/ultrastar-play.github.io
